### PR TITLE
datepicker timezone issue on view layer fixed

### PR DESCRIPTION
**Pull request by barshat7**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js [hunk 1]
```diff
@@ -1146,7 +1146,20 @@ if (typeof window.DatePickerWidget === 'undefined') {
...
+        // test if format is 'yyyy-MM-dd'
+        const pattern = /^\d{4}-\d{2}-\d{2}$/;
+        return pattern.test(str);
+      }
       let currDate =  new Date(value);
+      /**
+       * We only test for 'yyyy-MM-dd' because timezone related issue only occurs when date is selected from calendar.
+       * And calendar only sets value in yyyy-MM-dd
+       */
+      if (isDateDefaultFormat(value)) {
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1159]: Date will always come in this format, why have you adding this check ? We don't support any other format in saveValue.

Comment (barshat7) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1159]: If a user manually enters some other date format from the input widget

Comment (rismehta) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1159]: For saveValue there is no formatting, you don't need this check

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js [hunk 2]
```diff
@@ -1119,10 +1119,6 @@ if (typeof window.DatePickerWidget === 'undefined') {
       if (this.#isEditValueOrDisplayValue(this.#widget.value)) {
         return this.#model.value; // if the widget has edit/display value thus method should return model value
       }
-      let dateVal = new Date(this.toString());
-      if (!isNaN(dateVal)) {
```
Comment (sakshi-arora1) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1123]: Why was this added earlier?

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js [hunk 3]
```diff
@@ -99,6 +99,12 @@
                     this.widgetObject.setValue(e.target.value);
                     this.setActive();
                 }, this.getWidget());
+                this.widgetObject.addEventListener('input', (e) => {
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js:102]: I am not sure how is this change related to the current issue.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js [hunk 4]
```diff
@@ -1194,6 +1197,9 @@ if (typeof window.DatePickerWidget === 'undefined') {
           case 'focus':
             handler(e);
             break;
+          case 'input':
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1200]: Why do you need this listener in the current CL? Please fix this as a separate JIRA

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js (hunk 3 above)
Comment (barshat7) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js:102]: This issue is not directly related to timezone issue, but when we fix the timezone, this issue manifests in one of the test case.
Reason being that we were not clearing the value of the input field when user emptied it. (We do that from calendar icon but I think this was missed when manual entry of date was implemented). 
https://github.com/adobe/aem-core-forms-components/blob/ac6f9b768aa6cd807cbd4b3a40af936aafb7c5d0/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js#L1010

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js (hunk 4 above)
Comment (barshat7) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js:1200]: PR will not pass if this is not fixed, or will have to comment out a test case

**Review by sakshi-arora1 — APPROVED** [review]

**Review by rismehta — APPROVED** [review]
